craco.config.js

```diff
@@ -0,0 +1,116 @@
+const TerserPlugin = require("terser-webpack-plugin");
+const OptimizeCSSAssetsPlugin = require("optimize-css-assets-webpack-plugin");
+const safePostCssParser = require("postcss-safe-parser");
+const isWsl = require("is-wsl");
+
+module.exports = function({ env }) {
+  // The values below are copied wholesale from the webpack.config.js that
+  // CRA uses to generate it's builds.
+  // The following values have been modified to tweak the minification steps
+  // to support jujulib and the check-auth redux middleware for thunks.
+  // Both jujulib and the check-auth middleware require that the function names
+  // remain intact so we need to tell Terser to not mangle those names.
+  // - mangle.reserved
+  return {
+    webpack: {
+      configure: (webpackConfig, { env, paths }) => {
+        const shouldUseSourceMap = process.env.GENERATE_SOURCEMAP !== "false";
+        const isEnvProduction = env === "production";
+        const isEnvProductionProfile =
+          isEnvProduction && process.argv.includes("--profile");
+        return Object.assign(webpackConfig, {
+          optimization: {
+            minimize: isEnvProduction,
+            minimizer: [
+              // This is only used in production mode
+              new TerserPlugin({
+                terserOptions: {
+                  parse: {
+                    // We want terser to parse ecma 8 code. However, we don't want it
+                    // to apply any minification steps that turns valid ecma 5 code
+                    // into invalid ecma 5 code. This is why the 'compress' and 'output'
+                    // sections only apply transformations that are ecma 5 safe
+                    // https://github.com/facebook/create-react-app/pull/4234
+                    ecma: 8
+                  },
+                  compress: {
+                    ecma: 5,
+                    warnings: false,
+                    // Disabled because of an issue with Uglify breaking seemingly valid code:
+                    // https://github.com/facebook/create-react-app/issues/2376
+                    // Pending further investigation:
+                    // https://github.com/mishoo/UglifyJS2/issues/2011
+                    comparisons: false,
+                    // Disabled because of an issue with Terser breaking valid code:
+                    // https://github.com/facebook/create-react-app/issues/5250
+                    // Pending further investigation:
+                    // https://github.com/terser-js/terser/issues/120
+                    inline: 2
+                  },
+                  mangle: {
+                    safari10: true,
+                    reserved: [
```

Comment:
This is the only value changed from the CRA config so it will make for easy CRA upgrades in the future by copying the top level `optimization` key wholesale and then adding this key.

Reply:
> This is the only value changed from the CRA config so it will make for easy CRA upgrades in the future by copying the top level `optimization` key wholesale and then adding this key.

Would it be worth documenting the process for upgrading CRA in the project's readme?

Reply:
Good call, Done.